From b395fecb5e6383491fbfea20fdc96f64e8987db6 Mon Sep 17 00:00:00 2001
Message-ID: <b395fecb5e6383491fbfea20fdc96f64e8987db6.1736267959.git.mjg@fedoraproject.org>
From: Michael J Gruber <mjg@fedoraproject.org>
Date: Tue, 7 Jan 2025 17:08:22 +0100
Subject: [PATCH] fix incompatible pointer types

gcc15 turns some warnings into errors, so fix at least those which break
compilation with gcc15.
---
 include/bibtool/pxfile.h | 2 +-
 main.c                   | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/bibtool/pxfile.h b/include/bibtool/pxfile.h
index a9a3dac..3d05bc1 100644
--- a/include/bibtool/pxfile.h
+++ b/include/bibtool/pxfile.h
@@ -42,7 +42,7 @@ extern FILE * px_fopen(
 	char  * mode,
 	char  **pattern,
 	char  **path,
-	int   (*show)()
+	int   (*show)(char *)
 #endif
 	);
 
diff --git a/main.c b/main.c
index 3808cd6..a889c2b 100644
--- a/main.c
+++ b/main.c
@@ -414,7 +414,7 @@ int main(argc,argv)				   /*			     */
 						   /* this database.	     */
   int	i;				   	   /*			     */
   bool	need_rsc = true;		   	   /*			     */
-  int	(*fct)();			   	   /* Function pointer	     */
+  int	(*fct)(struct rECORD *, struct rECORD *);  /* Function pointer	     */
   int	c_len;					   /*                        */
   int   *c = NULL;				   /*                        */
  						   /*                        */
-- 
2.48.0.rc1.253.g8da213e47f

